spec/controllers/api/v2/searches_controller_spec.rb

```diff
@@ -276,45 +274,54 @@
     end
 
     context 'when the search options are valid' do
-      let!(:search) { double(ApiI14ySearch, as_json: { foo: 'bar'}, modules: %w(I14Y)) }
+      let!(:search) { instance_double(ApiI14ySearch, as_json: { foo: 'bar' }, modules: %w[I14Y]) }
 
       before do
-        expect(Affiliate).to receive(:find_by_name).and_return(affiliate)
+        allow(Affiliate).to receive(:find_by_name).and_return(affiliate)
 
-        expect(ApiI14ySearch).to receive(:new).with(hash_including(query: 'api')).and_return(search)
-        expect(search).to receive(:run)
-        expect(SearchImpression).to receive(:log).with(search,
-                                                   'i14y',
-                                                   hash_including('query'),
-                                                   be_a_kind_of(ActionDispatch::Request))
+        allow(ApiI14ySearch).to receive(:new).with(hash_including(query: 'api')).and_return(search)
+        allow(search).to receive(:run)
+        allow(SearchImpression).to receive(:log).with(search,
+                                                      'i14y',
+                                                      hash_including('query'),
+                                                      be_a_kind_of(ActionDispatch::Request))
 
-        get :i14y, params: search_params
+        get :i14y, params: search_params.merge(audience: 'everyone',
+                                               content_type: 'article',
+                                               mime_type: 'application/pdf',
+                                               searchgov_custom1: 'custom1',
+                                               searchgov_custom2: 'custom2, customtwo',
+                                               searchgov_custom3: 'custom3')
       end
 
       it { is_expected.to respond_with :success }
 
       it 'passes the correct options to its ApiI4ySearch object' do
-        expect(assigns(:search_options).attributes).to include({
-          access_key: 'basic_key',
-          affiliate: affiliate,
-          enable_highlighting: true,
-          file_type: 'pdf',
-          filter: '2',
-          limit: 20,
-          next_offset_within_limit: true,
-          offset: 0,
-          query: 'api',
-          query_not: 'excluded',
-          query_or: 'alternative',
-          query_quote: 'barack obama',
-          sort_by: 'date'
-        })
+        expect(assigns(:search_options).attributes).to include({ access_key: 'basic_key',
```

Comment:
Only substantive code change in spec file.  Remaining copy rubocop/code climate changes added in SRCH-3614.